java/src/shared/model/game/Game.java

```diff
@@ -710,6 +710,40 @@ public void discardCards(int playerIndex, List<ResourceType> cards) throws Playe
     }
 
     /**
+     * Action - Player rolls the dice
+     * @param value
+     * @throws InvalidDiceRollException
+     */
+    @Override
+    public void rollNumber(int value) throws Exception {
+        //Is value a 7 - robber
+        if(value == 7){
+            //Go to discarding phase before robbing if any player has to discard
+            getPlayers().forEach(player ->{
+                if(player.canDiscardCards()){
```

Comment:
I have something similar that will be merged soon, works with the discarding function.